plugin/pkg/scheduler/algorithm/priorities/priorities.go

```diff
@@ -170,6 +170,77 @@ func (n *NodeLabelPrioritizer) CalculateNodeLabelPriority(pod *api.Pod, machines
 	return result, nil
 }
 
+// This is a reasonable size range of all container images. 90%ile of images on dockerhub drops into this range.
+const minImgSize int64 = 23000000   // min 23MB
+const maxImgSize int64 = 1000000000 // max 1000MB
+
+// ImageLocalityPriority is a priority function that favors nodes with requested containers' images of a Pod.
+// It will detect whether the requested images are present on a node, and then calculate a score
+// range from 0 to 10 based on the sum size of those images:
+// - If all the images are not present on a node, this node will be given the lowest priority.
+// - If some of the images are present on a node, the larger their sizes' sum, the higher the node's priority.
+func ImageLocalityPriority(pod *api.Pod, machinesToPods map[string][]*api.Pod, podLister algorithm.PodLister, nodeLister algorithm.NodeLister) (schedulerapi.HostPriorityList, error) {
+	var score int
+	var sumSizeMap = make(map[string]int64)
+
+	nodes, err := nodeLister.List()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, container := range pod.Spec.Containers {
+		for _, node := range nodes.Items {
+
+			// Check if this container's image is present and get its size.
+			imageSize := checkContainerImageOnNode(node, container)
+			// Add this size to the total result of this node.
+			sumSizeMap[node.Name] += imageSize
+		}
+	}
+
+	result := []schedulerapi.HostPriority{}
+	// score int - scale of 0-10
+	// 0 being the lowest priority and 10 being the highest,
+	for nodeName, sumSize := range sumSizeMap {
+		switch {
+		case sumSize == 0 || sumSize < minImgSize:
+			// score == 0 means non of the images required by this Pod is present on this node,
+			// or the image is present but it's too small to put into further consideration.
+			score = 0
+		// If existing images' total size is larger than max, just make it highest priority.
+		case sumSize >= maxImgSize:
+			score = 10
+		default:
+			score = calculateScoreFromSize(sumSize)
+		}
+		result = append(result, schedulerapi.HostPriority{Host: nodeName, Score: score})
+	}
+	return result, nil
+}
+
+// Check if a container image is present on a node, and return its size.
+func checkContainerImageOnNode(node api.Node, container api.Container) int64 {
+	for _, image := range node.Status.Images {
+		for _, repoTag := range image.RepoTags {
+			if container.Image == repoTag {
+				// Should return immediately.
+				return image.Size
+			}
+		}
+	}
+	return 0
+}
+
+// Calculate the priority of a node. sumSize is sum size of requested images on this node.
+// 1. Split image size range into 10 buckets.
+// 2. Decide the priority of a given sumSize based on which bucket it belongs to.
+func calculateScoreFromSize(sumSize int64) int {
+	// Number of elements in each bucket
+	elementNumbers := (maxImgSize - minImgSize + 1) / 10
+	// Return which bucket the given size belongs to
+	return int((sumSize-minImgSize)/elementNumbers + 1)
```

Comment:
Is there a reason why this function isn't as simple as:

`return int((10 * (sumSize - minImgSize) / (maxImgSize - minImgSize)) + 1)`

That's a more obvious and easily recognizable scaling expression IMO.